packages/core/src/utils/path.ts

```diff
@@ -7,3 +7,29 @@
 export function getDirFromFilePath(path: string) {
   return path.substring(0, Math.max(path.lastIndexOf('/'), path.lastIndexOf('\\')))
 }
+
+/**
+ * Combine multiple uri parts into a single uri taking into account slashes.
+ *
+ * @param parts the parts to combine
+ * @returns the combined url
+ */
+export function joinUriParts(base: string, parts: string[]) {
```

Comment:
Might be good to call `trim()` all strings before checking whether they end with a `/`, or start.